redfish/manager.go

```diff
@@ -322,9 +324,19 @@ func (manager *Manager) UnmarshalJSON(b []byte) error {
 		return err
 	}
 
+	// parse and retrieve all link objects
+	var allLinks struct {
+		Links map[string]interface{}
+	}
+	err = json.Unmarshal(b, &allLinks)
+	if err != nil {
+		return err
+	}
+
 	// Extract the links to other entities
 	*manager = Manager(t.temp)
 	manager.ethernetInterfaces = string(t.EthernetInterfaces)
+	manager.Links = allLinks.Links
```

Comment:
Links are already parsed below. Those are needed for getting the links to other resources.

This doesn't look like what you were looking for in the issue filed though. Based on that, you want to get links that are returned under the OEM section. That's a bit different than here. I haven't thought of a good, intuitive way to get OEM data since it could include links, strings, or other nested objects depending on each individual vendor.

So I don't believe this approach is right, but unfortunately I don't have a good answer for that right now.

Reply:
yes, I noticed it has parsed with explicit fields, but I have to double parse because of current implementation. Agree it is not elegant.

I'm thinking another two possible solutions:
 - solution 1: add LinkOEM field whose type is map[string]interface{}, to expose this section. This will be same as other field, and can embed into `type linkReference struct {` for parsing.
 - solution 2: have one Raw method to cache and return original whole section, and let caller parse if current data is not enough. This seems more generic. 

What do you think?

Reply:
I did play around with having an OEM field of map[string]interface{} once, but I seem to recall there being some issue with it with different returned OEM data. If you get something like that working, I think that would be great. We just need to make sure to try it against a few different vendor implementations to make sure it actually covers the different options.

I don't think LinkOEM is the right name for it though. Probably need an OEMData or OEMField(key) method seems like a better fit. Some of the returned values may be links, but most OEM data I've seen is actually string or integer values.

Reply:
The reason I proposes `LinkOEM` is due to Dell Manager body has 2 OEM sections at different places, such as below partial data. Let me know what would be the best name in this case? Once we had the name, I'll test the patch against both HPE and DELL.

```
{
  "@odata.context": "/redfish/v1/$metadata#Manager.Manager",
  "@odata.id": "/redfish/v1/Managers/iDRAC.Embedded.1",
  "@odata.type": "#Manager.v1_6_0.Manager",
  "Id": "iDRAC.Embedded.1",
  "Links": {
    "Oem": {
      "Dell": {
        "DellAttributes": [
          {
            "@odata.id": "/redfish/v1/Managers/iDRAC.Embedded.1/Attributes"
          },
          {
            "@odata.id": "/redfish/v1/Managers/System.Embedded.1/Attributes"
          },
          {
            "@odata.id": "/redfish/v1/Managers/LifecycleController.Embedded.1/Attributes"
          }
        ],
        "DellAttributes@odata.count": 3,
        "DellJobService": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellJobService"
        },
        "DellLCService": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellLCService"
        },
        "DellLicensableDeviceCollection": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellLicensableDeviceCollection"
        },
        "DellLicenseCollection": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellLicenseCollection"
        },
        "DellLicenseManagementService": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellLicenseManagementService"
        },
        "DellOpaqueManagementDataCollection": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellOpaqueManagementDataCollection"
        },
        "DellPersistentStorageService": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellPersistentStorageService"
        },
        "DellSwitchConnectionCollection": {
          "@odata.id": "/redfish/v1/Dell/Systems/System.Embedded.1/NetworkPorts/DellSwitchConnectionCollection"
        },
        "DellSwitchConnectionService": {
          "@odata.id": "/redfish/v1/Dell/Systems/System.Embedded.1/DellSwitchConnectionService"
        },
        "DellSystemManagementService": {
          "@odata.id": "/redfish/v1/Dell/Systems/System.Embedded.1/DellSystemManagementService"
        },
        "DellSystemQuickSyncCollection": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellSystemQuickSyncCollection"
        },
        "DellTimeService": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellTimeService"
        },
        "DellUSBDeviceCollection": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellUSBDeviceCollection"
        },
        "DelliDRACCardService": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DelliDRACCardService"
        },
        "DellvFlashCollection": {
          "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellvFlashCollection"
        },
        "Jobs": {
          "@odata.id": "/redfish/v1/Managers/iDRAC.Embedded.1/Jobs"
        }
      }
    }
  },
  "Oem": {
    "Dell": {
      "DelliDRACCard": {
        "@odata.context": "/redfish/v1/$metadata#DelliDRACCard.DelliDRACCard",
        "@odata.id": "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DelliDRACCard/iDRAC.Embedded.1-1_0x23_IDRACinfo",
        "@odata.type": "#DelliDRACCard.v1_1_0.DelliDRACCard",
        "IPMIVersion": "2.0",
        "LastSystemInventoryTime": "2020-10-19T19:32:53+00:00",
        "LastUpdateTime": "2020-10-19T21:43:41+00:00",
        "URLString": "https://10.5.1.83:443"
      }
    }
  },
  "PowerState": "On",
  "Status": {
    "Health": "OK",
    "State": "Enabled"
  },
  "UUID": "3330484f-c0c8-3180-5410-00544c4c4544"
}
```

Reply:
Oh, good point. I was thinking of the latter case, but you are correct - there is Manager.Oem and Manager.Links.Oem. So extracting the OEM links out to something accessible should be good. I would suggest OEMLinks rather than LinksOEM.

Reply:
Sounds good. I will make two fields: `OEMLinks` and `OEMData`